week4/spiral/jasonkuhrt/Main.hs

```diff
@@ -0,0 +1,153 @@
+{- README
+
+-- Sprial
+
+--- Spec
+
+Given the width of a spiral, draw a spiral. For example:
+
+    > spiral 9
+
+    *********
+            *
+    ******* *
+    *     * *
+    * *** * *
+    * * * * *
+    * *   * *
+    * ***** *
+    *       *
+    *********
+
+
+
+-- Problem Solving
+
+    2 4 5 3 1 <---- Step
+                       |
+            8 <- Size  |
+    6     4 |       |  |
+    | 2   | |       |  |
+    | | 1 | |       |  |
+    | | | | |       |  |
+    ********* ----- 9  1
+          B *
+    ******* * ---- 7   3
+    *   D * *
+    * *** * * -- 3     5
+    * * * * *
+    * * E * *
+    * ***** * --- 5    4
+    * C     *
+    ********* ----- 9  2
+    A
+
+     !      A    B    C    D    E
+(11) 9,8    9,6  7,4  5,2  3,0  1,0
+
+Specifically:
+
+                    - > 9 7 5 3 1 0
+                    | > 6 4 2 0 0 0
+      !
+     -9 *********
+                *    |6
+                *                -7 *******
+                *    *                    *    |2
+                *    *                    *            -3 ***
+                *    *                    *     *          |0   -1 *
+                *    *                    *     *
+                *    *                          ***** -5
+                *    *                   |4
+                     ********* -9
+               |8
+
+* H/V lengths shrink by 2 each step
+* V length is H - 3
+* Negative values are ignored
+
+! NOTE First loop is special case because it doesn't nest (h is smaller by 1)
+
+TODO Given the above pattern there should be a simple calculation to solve this problem...
+-}
+
+
+
+module Main where
+
+import Data.Matrix (Matrix)
+import qualified Data.Matrix as Matrix
+
+
+
+main :: IO ()
+main = do
+  putStrLn "How wide do you want the spiral to be?"
+  putStrLn . render . spiral . (read :: String -> Int) =<< getLine
```

Comment:
Hm! I don't remember now. I think I might have copied something I tested in `ghci` where it was necessary? At any rate I just revised and tested and yup its not needed. Thanks.